src/main/java/de/craften/ui/swingmaterial/MaterialFrameWrapper.java

```diff
@@ -0,0 +1,372 @@
+
+                @Override
+                public void mousePressed(MouseEvent e) {
+                    if (e.getButton() == MouseEvent.BUTTON1
+                        && (control.getFrameState() & Frame.MAXIMIZED_BOTH) != Frame.MAXIMIZED_BOTH) {
+                        moving = true;
+                        movs[1][0] = e.getXOnScreen();
+                        movs[1][1] = e.getYOnScreen();
+                    }
+                }
+
+                @Override
+                public void mouseReleased(MouseEvent e) {
+                    if (e.getButton() == MouseEvent.BUTTON1
+                        && (control.getFrameState() & Frame.MAXIMIZED_BOTH) != Frame.MAXIMIZED_BOTH) {
+                        moving = false;
+                    }
+                }
+
+                @Override
+                public void mouseDragged(MouseEvent e){
+                    if (moving) {
+                        System.arraycopy(movs[1], 0, movs[0], 0, 2);
+                        movs[1][0] = e.getXOnScreen();
+                        movs[1][1] = e.getYOnScreen();
+                        int mx = movs[1][0] - movs[0][0];
+                        int my = movs[1][1] - movs[0][1];
+                        if (mx != 0 || my != 0) {
+                            frame.setLocation(frame.getX() + mx, frame.getY() + my);
+                        }
+                    }
+                }
+            };
+        }
+    }
+    
+    /**
+     * A {@code FrameControler} provides an easy way to perform operations that
+     * change the window state and size as minimizing, maximizing and restoring.
+     */
+    protected static class FrameControler {
+        /** The {@link MaterialFrame} being controled. */
+        MaterialFrame frame;
+
+        protected FrameControler(MaterialFrame frame) {
+            this.frame = frame;
+        }
+        
+        /**
+         * This method iconifies (minimizes) a frame into the task bar. The
+         * maximized bits are not affected.
+         */
+        public void iconify() {
+            int state = frame.getExtendedState();
+            // Set the iconified bit
+            state |= Frame.ICONIFIED;
+            // Iconify the frame
+            frame.setExtendedState(state);
+        }
+
+        /**
+         * This method deiconifies a frame from the task bar. The maximized bits
+         * are not affected.
+         */
+        public void deiconify() {
+            int state = frame.getExtendedState();
+            // Clear the iconified bit
+            state &= ~Frame.ICONIFIED;
+            // Deiconify the frame
+            frame.setExtendedState(state);
+        }
+
+        /**
+         * This method restores a frame from a maximized state. The iconified
+         * bit is not affected.
+         */
+        public void restore() {
+            int state = frame.getExtendedState();
+            // Clear the maximized bits
+            state &= ~Frame.MAXIMIZED_BOTH;

```

Comment:
**Short answer:** This is.

**Long answer:** That's exactly what `FrameControler` is for, which code you're looking right now. Using bit arithmetic is exactly what the Swing docs suggest to do (at least they did last time I checked them out).